opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+
+        # assign a local folder 'synthetic_queries/' to store the unzip file,
+        # check if the folder contains sub-folders and files, remove and clean up the folder before unzip.
+        # walk through the zip file and read the file paths into file_list
+        unzip_path = os.path.join(os.getcwd(), "synthetic_queries/")
+
+        # ML add warning here and confirm with user to proceed
+        if os.path.exists(unzip_path):
+            if len(os.listdir(unzip_path)) > 0:
+                for files in os.listdir(unzip_path):
+                    sub_path = os.path.join(unzip_path, files)
+                    try:
+                        shutil.rmtree(sub_path)
+                    except OSError as err:
+                        print(
+                            "Fail to delete files, please delete all files in "
+                            + str(unzip_path)
+                            + " "
+                            + str(err)
+                        )
+
+        with ZipFile(read_path, "r") as zip_ref:
+            zip_ref.extractall(unzip_path)
+
+        for root, dirnames, filenames in os.walk(unzip_path):
+            for filename in filenames:
+                file_list.append(os.path.join(root, filename))
+
+        # check empty zip file
+        num_file = len(file_list)
+
+        if num_file == 0:
+            raise Exception(
+                "Zipped file is empty. Please provide a zip file with nonzero synthetic queries."
+            )
+
+        for file_path in file_list:
+            f = open(file_path, "rb")
+            print("reading synthetic query file: " + file_path)
+            process.append(pickle.load(f))
+            f.close()
+
+        # reading the files to get the probability, queries and passages
+        prob = []
+        query = []
+        passages = []
+        for j in range(0, num_file):
+            for dict_str in process[j]:
+                if "probability" in dict_str.keys():
+                    prob.append(dict_str["probability"])
+                if "passage" in dict_str.keys():
+                    passages.append(dict_str["passage"])
+                if "query" in dict_str.keys():
+                    query.append(dict_str["query"])
+
+        df = pd.DataFrame(
+            list(zip(prob, query, passages)), columns=["prob", "query", "passages"]
+        )
+
+        df = df.drop_duplicates(subset=["query"])
+        df["passages"] = df.apply(lambda x: self.qryrem(x), axis=1)
+
+        df = df.sample(frac=1)
+        return df
+
+    def load_sentence_transformer_example(
+        self, df, use_accelerate: bool = False
+    ) -> List[str]:
+        """
+        Description:
+        Create input data for training
+
+        Parameters:
+        df: pd.DataFrame
+            required for loading sentence transformer examples.
+        use_accelerate: bool = False,
+            Optional, use accelerate to fine tune model. Default as false to not use accelerator.
```

Comment:
This description is not quite clear. What happens if `use_accelerate` is false? it won't fine tune the model? Describe clearly the purpose of using this variable.

Reply:
added comment for accelerate.

Reply:
added in 
[added save_as_pt function](https://github.com/opensearch-project/opensearch-py-ml/pull/18/commits/094339cca0a561d3ecd4edf74ce803b77e973239)